modules/core/src/loaders/xviz-file-loader.js

```diff
@@ -73,26 +73,27 @@ export default class XVIZFileLoader extends XVIZLoaderInterface {
     return this._loadFile(metadataPath, {worker: false});
   }
 
-  _loadNextBatch(startFrame) {
+  _startLoad(startFrame) {
+    // fetching in parallel
+    for (let i = 0; i < this._batchSize && startFrame + i < this._numberOfFrames; i++) {
+      this._load(startFrame + i);
+    }
+  }
+
+  _load(frame) {
```

Comment:
The order of frames being fetched are not guaranteed.

Reply:
You can store the last loaded frame in the class. Then instead of calling `_load(index)` you just call `_loadNextFrame()` whenever a spot is freed.